ctp2_code/ui/aui_common/aui_pixel.cpp

```diff
@@ -73,7 +73,7 @@ uint8 aui_Pixel::GetPaletteIndexedColor( uint8 red, uint8 green, uint8 blue, RGB
 	sint32  diff    = INT_MAX;
 	uint16  valMain = ColorCode555(red, green, blue);
 
-	for (uint8 i = 0; i < 256; i++)
```

Comment:
Would this case an endless loop or did it accidentally work?

Reply:
As a uint8 i will always be less than 256, but this will not loop endlessly as it will find a colour match, at which point it exits the loop. This change only fixes an issue because the compiler detects that checking against 256 is incorrect.

I guess the only execution problem that could even potentially occur would be if the compiler tried to demote 256 to unit8 do the check, rather than promoting i to int, in which case 256 would become 0 and the loop would never execute. I don't think any compiler would actually be allowed to do that, though.